_posts/2018-02-12-continuous-releases.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+tags: development-process
+date: 2018-02-12 13:14
+title: The Hypothesis continuous release process
+published: true
+author: alexwlchan
+---
+
+If you watch [the Hypothesis changelog][changelog], you'll notice the rate of releases sped up dramatically in 2017.
+We released over a hundred different versions, sometimes multiple times a day.
+
+This is all thanks to our continuous release process.
+We've completely automated the process of releasing, so every pull request that changes code gets a new release, without any human input.
+In this post, I'll explain how our continuous releases work, and why we find it so useful.
+
+[changelog]: https://hypothesis.readthedocs.io/en/latest/changes.html
+
+<!--more-->
+
+## How it works
+
+In the past, Hypothesis was released manually.
+Somebody had to write a changelog, tag a new release on GitHub, and run some manual pip commands to publish a new version to PyPI -- and only David had the credentials for the latter.
+
+This meant that releases were somewhat infrequent, and features spent a long time in master before they were available to `pip install`.
+The pace of development picked up in 2017 -- partly as new maintainers arrived, and partly work for David's PhD -- and we wanted to be able to release more frequently.
```

Comment:
Maybe clarify that the PhD hadn't actually started then? e.g. "groundwork for David's upcoming (now started) PhD"? (Not at all wed to that wording)